doozer/doozerlib/backend/rebaser.py

```diff
@@ -0,0 +1,1549 @@
+from doozerlib.brew import BuildStates
+from doozerlib.image import ImageMetadata
+from doozerlib.record_logger import RecordLogger
+from doozerlib.repos import Repos
+from doozerlib.runtime import Runtime
+from doozerlib.source_modifications import SourceModifierFactory
+from doozerlib.source_resolver import SourceResolution, SourceResolver
+
+LOGGER = logging.getLogger(__name__)
+
+
+CONTAINER_YAML_HEADER = """
+# This file is managed by Doozer: https://github.com/openshift-eng/art-tools/tree/main/doozer
+# operated by the OpenShift Automated Release Tooling team (#forum-ocp-art on CoreOS Slack).
+
+# Any manual changes will be overwritten by Doozer on the next build.
+#
+# See https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/odcs_integration_with_osbs
+# for more information on maintaining this file and the format and examples
+
+---
+"""
+
+
+# Doozer used to be part of OIT
+OIT_COMMENT_PREFIX = '#oit##'
+OIT_BEGIN = '##OIT_BEGIN'
+OIT_END = '##OIT_END'
+
+
+class KonfluxRebaser:
+    """ Rebase images to a new branch in the build source repository.
+
+    FIXME: Some features that are known to be missing:
+    - Handling of alternate upstreams
+    - Handling of private fix bit in parent images
+    """
+
+    def __init__(self,
+                 runtime: Runtime,
+                 base_dir: Path,
+                 source_resolver: SourceResolver,
+                 upcycle: bool = False, force_private_bit: bool = False,
+                 record_logger: Optional[RecordLogger] = None,
+                 source_modifier_factory=SourceModifierFactory(),
+                 logger: Optional[logging.Logger] = None,
+                 ) -> None:
+        self._runtime = runtime
+        self._base_dir = base_dir
+        self._source_resolver = source_resolver
+        self.upcycle = upcycle
+        self.force_private_bit = force_private_bit
+        self._record_logger = record_logger
+        self._source_modifier_factory = source_modifier_factory
+        self.should_match_upstream = False  # FIXME: Matching upstream is not supported yet
+        self._logger = logger or LOGGER
+
+    async def rebase_to(
+            self,
+            metadata: ImageMetadata,
+            version: str,
+            input_release: str,
+            force_yum_updates: bool,
+            commit_message: str,
+            push: bool,
+    ) -> None:
+        # If this image has an upstream source, resolve it
+        source = None
+        if metadata.has_source():
+            self._logger.info(f"Resolving source for {metadata.qualified_key}")
+            source = cast(SourceResolution, await exectools.to_thread(self._source_resolver.resolve_source, metadata))
+        else:
+            raise IOError(f"Image {metadata.qualified_key} doesn't have upstream source. This is no longer supported.")
+
+        dest_branch = "art-{group}-assembly-{assembly_name}-dgk-{distgit_key}".format_map({
```

Comment:
does it make sense to have `distgit_key`? since there will be no "distgit" in konflux? maybe image_repo_key or image_key or similar? Just a suggestion maybe this requires a wider refactor.

Reply:
You are right that the variable name `distgit_key` is misleading. We discussed this on a previous team call and most people agreed that distgit_key is a well understood concept and prefer to keep that. This is already documented on https://docs.google.com/document/d/1GK4iS55KEW-qpeMp6Xia2g88WvN4nVOBTwprim3swBU/edit#heading=h.9na33vz78zfa.